classes/class4/homework.md

```diff
@@ -0,0 +1,113 @@
+and you have 1000 bookstores which stock the book. That means you have to go update 1000 objects every hour,
+compared to 1 per hour if you had used referencing.
+This becomes an even bigger problem when you're a product like Twitter and your
+"bookstores" are users and books are people those users follow. Let's say you want
+to update information about the book "Lady Gaga", which is stocked by 33 million "bookstores".
+This would be nearly impossible with embedded data, but is a synch with references.
+
+In the end which way you use (reference or embedding) depends what your data access patterns will be like.
+You'll likely be using embeds 80% of the time, but references also have their place, so know how to do both.
+
+The Mongo documentation has further details about [when to embed vs reference](http://docs.mongodb.org/manual/core/data-model-design/).
+
+## Assignment
+
+Great news! We started shopping around your last two homeworks (hope you don't mind)
+and although no one was interested in buying your cat tracking software,
+we did get an email from a local burger joint, looking for some help.
+Jessica's Burgers is looking to update their aging ordering system to the 21st century.
+So in an effort to ~~make us loads of cash~~ improve your coding skills,
+this homework will focus on making a web app which will help Jessica's customers
+get their delicious burgers even quicker (and more delicously).
+Your application will allow users to build orders for single burger from a list of ingredients
+(which will need to be updated as new stock arrives).
+Then it will allow Jessica's chefs to see all the pending orders, fill them, and alert customers that their burger is ready.
+
+Your application will need the following http endpoints:
+* `/ingredients` =>
+  * Shows a list of current ingredients (Name and Price) with delete and edit button.
+  * Delete button will tell the server to remove the ingredient for new orders (???).
+  The ingredient should be removed from the current page without refreshing.
+  * Edit button allows the user to submit a new name or price for the ingredient
+  which the server will update. The edits should change the ingredient list without refreshing.
+* `/order` =>
+  * Shows a form which allows customers to create a new burger.
+  * There should be a checklist of all available ingredients and their price.
+  * There should a Submit button that will send the server the new order without refreshing the page.
+  You should give your customer a nice congratulatory message for completing their order
+  (maybe a free cat picture since you're so good at that?!)
+  * A running counter of total cost: Should update whenever a new ingredient is added or removed.
+* `/kitchen` =>
+  * Shows a list of all pending orders.
+  * A `completed` button beside each order that tells the server the order is complete.
+  Clicking this should remove the order from the list of orders without refreshing the page.
```

Comment:
I think they didn't have as much in-class content (like an Express tutorial). We should be fine without it.